reana_server/status.py

```diff
@@ -341,12 +361,285 @@ def get_status(self):
         }
 
 
+class NodesStatus(REANAStatus):
+    """Class to retrieve statistics related to REANA cluster nodes."""
+
+    def get_nodes(self):
+        """Get list of all node names."""
+        nodes = current_k8s_corev1_api_client.list_node()
+        return [node.metadata.name for node in nodes.items]
+
+    def get_unschedulable_nodes(self):
+        """Get list of node names that are not schedulable."""
+        nodes = current_k8s_corev1_api_client.list_node(
+            field_selector="spec.unschedulable=true"
+        )
+        return [node.metadata.name for node in nodes.items]
+
+    def get_memory_usage(self):
+        """Get nodes memory usage."""
+        result = dict()
+        nodes = current_k8s_corev1_api_client.list_node()
+        for node in nodes.items:
+            result[node.metadata.name] = {"capacity": node.status.capacity["memory"]}
+
+        try:
+            node_metrics = current_k8s_custom_objects_api_client.list_cluster_custom_object(
+                "metrics.k8s.io", "v1beta1", "nodes"
+            )
+            for node_metric in node_metrics.get("items", []):
+                node_name = node_metric["metadata"]["name"]
+                result[node_name]["usage"] = node_metric["usage"]["memory"]
+
+                node_capacity = result[node_name]["capacity"]
+                node_usage = result[node_name]["usage"]
+                node_usage_percentage = round(
```

Comment:
Minor: probably `ClusterHealth.get_percentage()` could be utilized here? Also maybe `get_percentage()` could be moved to `utils.py`? I see we have something similar there `get_usage_percentage()`, but it returns a string instead of int..